packages/@react-stately/layout/src/ListLayout.ts

```diff
@@ -278,15 +274,15 @@ export class ListLayout<T> extends Layout<Node<T>, ListLayoutProps> implements K
     );
   }
 
-  buildChild(node: Node<T>, x: number, y: number): LayoutNode {
+  protected buildChild(node: Node<T>, x: number, y: number, parentKey: Key | null): LayoutNode {
```

Comment:
This now takes a parentKey as a parameter rather than using the node's parentKey in order to properly handled TreeView's flattened collection. We want the parent LayoutInfo here, not the parent Node.